js/src/modal.js

```diff
@@ -109,6 +110,15 @@ const Modal = (($) => {
     }
 
     show(relatedTarget) {
+      if (this._isTransitioning) {
+        throw new Error('Modal is transitioning')
+      }
+
+      let transition = Util.supportsTransitionEnd() &&
+        $(this._element).hasClass(ClassName.FADE)
+      if (transition) {
+        this._isTransitioning = true
```

Comment:
We should only enter the transitioning state after line 130, because we won't actually try to show the modal if the `show` event gets preventDefault-ed (see lines 126-130).